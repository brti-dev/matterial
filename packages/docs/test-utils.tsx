import React, { FC, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

function ThemeProvider({ theme, children }) {
  return <>{children}</>
}

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme="light">{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
