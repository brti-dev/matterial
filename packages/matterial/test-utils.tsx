import { FC, ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

// Example of wrapping a test
function ThemeProvider({ theme, children }: any) {
  return <div className={`theme--${theme}`}>{children}</div>
}

const AllTheProviders: FC = ({ children }: any) => {
  return <ThemeProvider theme="light">{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
