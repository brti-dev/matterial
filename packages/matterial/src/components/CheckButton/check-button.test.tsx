import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'

import { render, screen } from '../../test-utils'
import { CheckButton, CheckButtonGroup } from '.'

const CheckButtons = ({ checked, setChecked = () => {}, ...rest }: any) => (
  <>
    <CheckButton
      name="daddy"
      value="true"
      checked={checked == 'daddy'}
      onChange={() => setChecked('daddy')}
      {...rest}
    >
      I'm a daddy
    </CheckButton>
    <CheckButton
      name="has_boat"
      value="true"
      checked={checked == 'has_boat'}
      onChange={() => setChecked('has_boat')}
      {...rest}
    >
      I have a boat
    </CheckButton>
  </>
)

test('should render correctly', () => {
  const tree = renderer
    .create(
      <CheckButtonGroup orientation="vertical" className="cb__group">
        <CheckButtons checked="daddy" className="cb__item" />
      </CheckButtonGroup>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

describe('check button', () => {
  test('should be checked when set as `checked`', () => {
    const { getAllByRole } = render(
      <CheckButtonGroup>
        <CheckButtons checked="daddy" />
      </CheckButtonGroup>
    )

    const checkBox = getAllByRole('checkbox')[0]
    expect(checkBox).toBeChecked()
  })

  test('should execute callback when clicked', () => {
    let checked = false
    render(
      <CheckButtonGroup>
        <CheckButtons setChecked={() => (checked = true)} />
      </CheckButtonGroup>
    )

    userEvent.click(screen.getByText("I'm a daddy"))
    expect(checked).toBeTruthy()
  })
})

describe('check button group', () => {
  test('should render vertically', () => {
    const id = 'check-button-group'

    const { getByTestId } = render(
      <CheckButtonGroup orientation="vertical" data-testid={id}>
        <CheckButtons />
      </CheckButtonGroup>
    )

    expect(getByTestId(id)).toHaveClass('containerVertical')
  })

  test('should render as an overloaded element', () => {
    const id = 'check-button-group'
    const Row = ({ id }: { id: string }) => (
      <span data-testid={id} className="foo" />
    )

    const { getByTestId } = render(
      <CheckButtonGroup className="bar" as={Row} id={id}>
        <CheckButtons />
      </CheckButtonGroup>
    )

    expect(getByTestId(id)).toHaveClass('foo')
  })
})
