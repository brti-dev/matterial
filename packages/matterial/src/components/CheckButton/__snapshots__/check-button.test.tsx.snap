// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="cb__group container containerVertical"
>
  <div
    className="cb__item checkButton"
  >
    <input
      checked={true}
      className="visually-hidden"
      id="checkButton__daddy__true"
      name="daddy"
      onChange={[Function]}
      type="checkbox"
      value="true"
    />
    <label
      className="button color--default size--medium variant--default button variant--outlined checked"
      htmlFor="checkButton__daddy__true"
      style={
        Object {
          "--color": "var(--color-default)",
        }
      }
      type="button"
    >
      <span
        className="main-content"
      >
        I'm a daddy
      </span>
    </label>
  </div>
  <div
    className="cb__item checkButton"
  >
    <input
      checked={false}
      className="visually-hidden"
      id="checkButton__has_boat__true"
      name="has_boat"
      onChange={[Function]}
      type="checkbox"
      value="true"
    />
    <label
      className="button color--default size--medium variant--default button variant--outlined"
      htmlFor="checkButton__has_boat__true"
      style={
        Object {
          "--color": "var(--color-default)",
        }
      }
      type="button"
    >
      <span
        className="main-content"
      >
        I have a boat
      </span>
    </label>
  </div>
</div>
`;
